great_expectations/checkpoint/checkpoint.py

```diff
@@ -200,6 +199,20 @@ def _build_result_key(
             batch_identifier=batch_identifier,
         )
 
+    def _construct_result(
+        self,
+        run_id: RunIdentifier,
+        run_results: Dict[ValidationResultIdentifier, ExpectationSuiteValidationResult],
+    ) -> CheckpointResult:
+        for result in run_results.values():
+            result.meta["checkpoint_id"] = self.id
```

Comment:
Non-blocking, but I'm still in favor of passing checkpoint id down to where results are created and just fully create the result in one place.

Reply:
Sounds great